electrum/interface.py

```diff
@@ -404,6 +406,9 @@ async def wrapper_func(self: 'Interface', *args, **kwargs):
     @log_exceptions
     @handle_disconnect
     async def run(self):
+        expected_fingerprint = self.network.config.get("fingerprint")
+        if expected_fingerprint and not await self.verify_server_certificate(expected_fingerprint):
+            raise ErrorSSLCertFingerprintMismatch('Refusing to connect to server due to SSL certificate fingerprint mismatch')
```

Comment:
if `self.protocol` is not "s", you should not run ssl-related code, e.g. `self.get_certificate()`.
Instead, if protocol is not "s" but fingerprint is provided, you could just fail with a distinct error.

Reply:
Implemented the check in https://github.com/spesmilo/electrum/pull/6094/commits/6d7698ec0b5d9078fb8c2e0c9dd5c334925e7533 but having the same issue I mentioned in the original comment with the error being caught somewhere instead of causing the application to exit.

Reply:
This is now resolved after refactoring the fingerprint check to be next to the other certificate related logic.